internal/impl/elasticsearchv8/output.go

```diff
@@ -0,0 +1,373 @@
+		Description(`
+Both the `+"`id` and `index`"+` fields can be dynamically set using function interpolations described xref:configuration:interpolation.adoc#bloblang-queries[here]. When sending batched messages these interpolations are performed per message part.`+service.OutputPerformanceDocs(true, true)).
+		Fields(
+			service.NewStringListField(esFieldURLs).
+				Description("A list of URLs to connect to. If an item of the list contains commas it will be expanded into multiple URLs.").
+				Example([]string{"http://localhost:9200"}),
+			service.NewInterpolatedStringField(esFieldIndex).
+				Description("The index to place messages."),
+			service.NewInterpolatedStringField(esFieldAction).
+				Description("The action to take on the document. This field must resolve to one of the following action types: `index`, `update` or `delete`."),
+			service.NewInterpolatedStringField(esFieldID).
+				Description("The ID for indexed messages. Interpolation should be used in order to create a unique ID for each message.").
+				Example(`${!counter()}-${!timestamp_unix()}`),
+			service.NewInterpolatedStringField(esFieldPipeline).
+				Description("An optional pipeline id to preprocess incoming documents.").
+				Advanced().
+				Default(""),
+			service.NewInterpolatedStringField(esFieldRouting).
+				Description("The routing key to use for the document.").
+				Advanced().
+				Default(""),
+			service.NewIntField(esFieldRetryOnConflict).
+				Description("Specify how many times should an update operation be retried when a conflict occurs").
+				Advanced().
+				Default(0),
+			service.NewTLSToggledField(esFieldTLS),
+			service.NewOutputMaxInFlightField(),
+		).
+		Fields(
+			service.NewObjectField(esFieldAuth,
+				service.NewBoolField(esFieldAuthEnabled).
+					Description("Whether to use basic authentication in requests.").
+					Default(false),
+				service.NewStringField(esFieldAuthUsername).
+					Description("A username to authenticate as.").
+					Default(""),
+				service.NewStringField(esFieldAuthPassword).
+					Description("A password to authenticate with.").
+					Default("").Secret(),
+			).Description("Allows you to specify basic authentication.").
+				Advanced().
+				Optional(),
+			service.NewBatchPolicyField(esFieldBatching),
+		).
+		Example("Updating Documents", "When updating documents, the request body should contain a combination of a `doc`, `upsert`, and/or `script` fields at the top level, this should be done via mapping processors. `doc` updates using a partial document, `script` performs an update using a scripting language such as the built in Painless language, and `upsert` updates an existing document or inserts a new one if it doesn’t exist. For more information on the structures and behaviors of these fields, please see the https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html[Elasticsearch Update API^]", `
```

Comment:
Can we get a couple of solid examples? Maybe Redpanda -> elasticsearch_v8 and s3 -> elasticsearch_v8?

Reply:
Also can we add the `update` docs to either the `action` field or the overall summary? That's pretty important.

Reply:
Done, lmk what you think.